receiver/kafkareceiver/kafka_receiver_test.go

```diff
@@ -111,11 +114,22 @@ func TestTracesConsumerGroupHandler(t *testing.T) {
 
 	obsrecv, err := receiverhelper.NewObsReport(receiverhelper.ObsReportSettings{ReceiverCreateSettings: receivertest.NewNopSettings(metadata.Type)})
 	require.NoError(t, err)
+	var called atomic.Bool
 	c := tracesConsumerGroupHandler{
-		unmarshaler:      newPdataTracesUnmarshaler(&ptrace.ProtoUnmarshaler{}, defaultEncoding),
-		logger:           zap.NewNop(),
-		ready:            make(chan bool),
-		nextConsumer:     consumertest.NewNop(),
+		unmarshaler: newPdataTracesUnmarshaler(&ptrace.ProtoUnmarshaler{}, defaultEncoding),
+		logger:      zap.NewNop(),
+		ready:       make(chan bool),
+		nextConsumer: func() consumer.Traces {
+			c, err := consumer.NewTraces(func(ctx context.Context, _ ptrace.Traces) error {
+				defer called.CompareAndSwap(false, true)
```

Comment:
```suggestion
				defer called.Bool(true)
```

nit, don't think we particularly care if it is false first